### PR TITLE
Mutation test

### PR DESCRIPTION
**Pull request by vakepati**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-283

### Change description ###

Added Mutation Testing coverage

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -127,11 +128,45 @@ jacocoTestReport {
...
+    'VOID_METHOD_CALLS',
+    'EMPTY_RETURNS',
+    'FALSE_RETURNS',
+    'NULL_RETURNS',
+    'PRIMITIVE_RETURNS',
+    'TRUE_RETURNS',
+    'REMOVE_CONDITIONALS'
+  ])
+}
+project.tasks['pitest'].group = "Verification"
```
Comment (doncem) [inline build.gradle:161]: we can move to task definition this one, i guess

File: build.gradle [hunk 2]
```diff
@@ -237,6 +272,10 @@ dependencies {
   testImplementation group: 'org.junit.platform', name: 'junit-platform-commons', version: versions.junitPlatform
   testImplementation group: 'org.junit.platform', name: 'junit-platform-engine', version: versions.junitPlatform
 
+  testCompile group: 'org.pitest', name: 'pitest', version: '1.5.1'
+  testCompile 'info.solidsoft.gradle.pitest:gradle-pitest-plugin:1.5.1'
+  testCompile 'org.codehaus.sonar-plugins:sonar-pitest-plugin:0.5'
```
Comment (doncem) [inline build.gradle:277]: do those dependencies involve something? don't see any changes in tests so perhaps jsut allowing some access. but still - not convinced atm

File: build.gradle [hunk 3]
```diff
@@ -127,11 +128,45 @@ jacocoTestReport {
...
+    'INVERT_NEGS',
+    'MATH',
+    'NEGATE_CONDITIONALS',
+    'VOID_METHOD_CALLS',
+    'EMPTY_RETURNS',
+    'FALSE_RETURNS',
+    'NULL_RETURNS',
+    'PRIMITIVE_RETURNS',
+    'TRUE_RETURNS',
+    'REMOVE_CONDITIONALS'
```
Comment (rkondratowicz) [inline build.gradle:158]: are those set on some kind of programme wide guidelines?

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,5 +1,6 @@
+#Thu Jun 11 00:17:22 BST 2020
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.4.1-all.zip
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.4.1-bin.zip
-zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
```
Comment (rkondratowicz) [inline gradle/wrapper/gradle-wrapper.properties:6]: not related, can be reverted

File: build.gradle (hunk 3 above)
Comment (rkondratowicz) [reply build.gradle:159]: if so, can you add a link to it?
Just wondering why those specific mutator are being used here and not the default set for example

Comment (vakepati) [reply build.gradle:159]: I used Pcq-backend  and Pit test Website as a reference,The reason for specifying  other than default mutators is that they will apply more mutations on the unit tests along with default mutations.May i know is there any harm by those mutations?

File: build.gradle [hunk 5]
```diff
@@ -127,11 +128,45 @@ jacocoTestReport {
 
 project.tasks['sonarqube'].dependsOn jacocoTestReport
 
+pitest {
+  group = "Verification"
+  pitestVersion.set('1.5.1')
```
Comment (rkondratowicz) [inline build.gradle:133]: is this needed?

File: build.gradle [hunk 6]
```diff
@@ -127,11 +128,45 @@ jacocoTestReport {
...
+  junit5PluginVersion.set('0.12')
+  testSourceSets = [sourceSets.test]
+  mainSourceSets = [sourceSets.main]
```
Comment (rkondratowicz) [inline build.gradle:136]: are those needed? these look like defaults

Comment (vakepati) [reply build.gradle:136]: yes ,other wise mutator does n't know which files to be mutated.

**Pull request closed without merging**